kolibri/core/assets/src/views/k-modal.vue

```diff
@@ -0,0 +1,312 @@
+    },
+    mounted() {
+      this.$nextTick(() => {
+        if (!this.$refs.modal.contains(document.activeElement)) {
+          this.focusModal();
+        }
+      });
+      window.addEventListener('focus', this.focusElementTest, true);
+      window.addEventListener('scroll', this.preventScroll, true);
+    },
+    updated() {
+      this.setContentSectionMaxHeight();
+    },
+    destroyed() {
+      window.removeEventListener('focus', this.focusElementTest, true);
+      window.removeEventListener('scroll', this.preventScroll, true);
+      // Wait for events to finish propagating before changing the focus.
+      // Otherwise the `lastFocus` item receives events such as 'enter'.
+      window.setTimeout(() => this.lastFocus.focus());
+    },
+    methods: {
+      setContentSectionMaxHeight: debounce(function() {
+        if (this.$refs.title && this.$refs.actions) {
+          this.maxContentHeight =
+            this.windowSize.height -
+            this.$refs.title.clientHeight -
+            this.$refs.actions.clientHeight -
+            32;
+        }
+      }, 100),
+      emitCancelEvent() {
+        // Emitted when the cancel button is clicked or the esc key is pressed
+        this.$emit('cancel');
+      },
+      emitEnterEvent() {
+        // Emitted when the submit button or the enter key is pressed
+        this.$emit('submit');
+      },
+      focusModal() {
+        this.$refs.modal.focus();
+      },
+      focusElementTest(event) {
+        // switching apps - not relevant
+        if (event.target === window) {
+          return;
+        }
+        // not sure when this would be true
+        if (!this.$refs.modal) {
+          return;
+        }
+        // focus has escaped the modal - put it back!
+        if (!this.$refs.modal.contains(event.target)) {
+          this.focusModal();
+        }
+      },
+      preventScroll(event) {
+        event.preventDefault();
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="stylus" scoped>
+
+  @require '~kolibri.styles.definitions'
+
+  .modal-overlay
+    position: fixed
+    top: 0
+    left: 0
+    width: 100%
+    height: 100%
+    background: rgba(0, 0, 0, 0.7)
+    transition: opacity 0.3s ease
+    background-attachment: fixed
+    z-index: 24
+
+  // TODO: margins for stacked buttons.
+  .modal
+    position: absolute
+    top: 50%
+    left: 50%
+    transform: translate(-50%, -50%)
+    background: $core-bg-light
+    overflow-y: auto
+    border-radius: $radius
+    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.33)
+    margin: 0 auto
+
+    &:focus
+      outline: none
+
+  .fade-enter-active, .fade-leave-active
+    transition: all 0.3s ease
+
+  .fade-enter, .fade-leave-active
+    opacity: 0
+
+  .title
+    margin: 0
+    padding: 24px
+
+  .content
+    padding: 0 24px
+    overflow-y: auto
+
+  .actions
+    text-align: right
+    padding: 24px
+    button
+      margin: 0
+
+  .actions button:last-of-type
+    margin-left: 16px
```

Comment:
recommend:

* remove the [built-in margin](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/buttons-and-links/buttons.styl#L30) on buttons
* add a new core component, `k-button-group`, which is responsible for adding margins between buttons